Py4JError: org.bdgenomics.adam.algorithms.consensus.ConsensusGenerator.fromKnowns does not exist in the JVM

```python
        readsPath = self.resourceFile("small.1.sam")
        variantsPath = self.resourceFile("small.vcf")

        ac = ADAMContext(self.ss)

        reads = ac.loadAlignments(readsPath)
        knownIndels = ac.loadVariants(variantsPath)

>       realigned = reads.realignIndels(knownIndels)

_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
bdgenomics/adam/rdd.py:1096: in realignIndels
    consensusModel = self.sc._jvm.org.bdgenomics.adam.algorithms.consensus.ConsensusGenerator.fromKnowns(knownIndels._jvmRdd)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <py4j.java_gateway.JavaClass object at 0x10795ffd0>, name = 'fromKnowns'

    def __getattr__(self, name):
        if name in ["__str__", "__repr__"]:
            raise AttributeError

        command = proto.REFLECTION_COMMAND_NAME +\
            proto.REFL_GET_MEMBER_SUB_COMMAND_NAME +\
            self._fqn + "\n" +\
            name + "\n" +\
            proto.END_COMMAND_PART
        answer = self._gateway_client.send_command(command)

        if len(answer) > 1 and answer[0] == proto.SUCCESS:
            if answer[1] == proto.METHOD_TYPE:
                return JavaMember(
                    name, None, proto.STATIC_PREFIX + self._fqn,
                    self._gateway_client)
            elif answer[1].startswith(proto.CLASS_TYPE):
                return JavaClass(
                    self._fqn + "$" + name, self._gateway_client)
            else:
                return get_return_value(
                    answer, self._gateway_client, self._fqn, name)
        else:
            raise Py4JError(
>               "{0}.{1} does not exist in the JVM".format(self._fqn, name))
E           Py4JError: org.bdgenomics.adam.algorithms.consensus.ConsensusGenerator.fromKnowns does not exist in the JVM
```

--- a/adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGenerator.scala
+++ b/adam-core/src/main/scala/org/bdgenomics/adam/algorithms/consensus/ConsensusGenerator.scala
@@ -80,7 +80,29 @@ object ConsensusGenerator {
   }
 
   /**
-   * Provides a generator to extract consensuses from a known set of INDELs.
+   * (Java-specific) Provides a generator to extract consensuses from a known set of INDELs.
+   *
+   * @param rdd The previously called INDEL variants.
+   * @return A consensus generator that looks at previously called INDELs.
+   */
+  def fromKnownIndels(rdd: VariantDataset): ConsensusGenerator = {
+    new ConsensusGeneratorFromKnowns(rdd.rdd, 0)
+  }
+
+  /**
+   * (Java-specific) Provides a generator to extract consensuses from a known set of INDELs.
+   *
+   * @param rdd The previously called INDEL variants.
+   * @param flankSize The number of bases to flank each known INDEL by.
+   * @return A consensus generator that looks at previously called INDELs.
+   */
+  def fromKnownIndels(rdd: VariantDataset,
+                      flankSize: java.lang.Integer): ConsensusGenerator = {
+    new ConsensusGeneratorFromKnowns(rdd.rdd, flankSize)
+  }
+
+  /**
+   * (Scala-specific) Provides a generator to extract consensuses from a known set of INDELs.
    *
    * @param rdd The previously called INDEL variants.
    * @param flankSize The number of bases to flank each known INDEL by. Default
@@ -98,13 +120,14 @@ object ConsensusGenerator {
    * @return A consensus generator that generates consensuses with several
    *   methods.
    */
+  @scala.annotation.varargs
   def union(generators: ConsensusGenerator*): ConsensusGenerator = {
     UnionConsensusGenerator(generators.toSeq)
   }
 }
 
 /**
- * Trait for generating consensus sequences for INDEL realignment.
+ * Abstract class for generating consensus sequences for INDEL realignment.
  *
  * INDEL realignment scores read alignments against the reference genome and
  * a set of "consensus" sequences. These consensus sequences represent alternate
@@ -113,7 +136,7 @@ object ConsensusGenerator {
  * trait provides an interface that a consensus generation method should
  * implement to provide it's consensus sequences to the realigner.
  */
-trait ConsensusGenerator extends Serializable {
+abstract class ConsensusGenerator extends Serializable {
 
   /**
    * @param cigar The CIGAR to process.

--- a/adam-python/bdgenomics/adam/rdd.py
+++ b/adam-python/bdgenomics/adam/rdd.py
@@ -1027,9 +1027,7 @@ class AlignmentRecordDataset(GenomicDataset):
                       maxReadsPerTarget = 20000,
                       unclipReads = False):
         """
-        Realigns indels using a concensus-based heuristic.
-
-        Generates consensuses from reads.
+        Realigns indels using a consensus-based heuristic from reads.
 
         :param bool isSorted: If the input data is sorted, setting this
         parameter to true avoids a second sort.
@@ -1060,19 +1058,17 @@ class AlignmentRecordDataset(GenomicDataset):
                                   self.sc)
 
 
-    def realignIndels(self,
-                      knownIndels,
-                      isSorted = False,
-                      maxIndelSize = 500,
-                      maxConsensusNumber = 30,
-                      lodThreshold = 5.0,
-                      maxTargetSize = 3000,
-                      maxReadsPerTarget = 20000,
-                      unclipReads = False):
+    def realignIndelsFromKnownIndels(self,
+                                     knownIndels,
+                                     isSorted = False,
+                                     maxIndelSize = 500,
+                                     maxConsensusNumber = 30,
+                                     lodThreshold = 5.0,
+                                     maxTargetSize = 3000,
+                                     maxReadsPerTarget = 20000,
+                                     unclipReads = False):
         """
-        Realigns indels using a concensus-based heuristic.
-
-        Generates consensuses from prior called INDELs.
+        Realigns indels using a consensus-based heuristic from prior called INDELs.
 
         :param bdgenomics.adam.rdd.VariantDataset knownIndels: An RDD of previously
         called INDEL variants.
@@ -1093,7 +1089,7 @@ class AlignmentRecordDataset(GenomicDataset):
         :rtype: bdgenomics.adam.rdd.AlignmentRecordDataset
         """
 
-        consensusModel = self.sc._jvm.org.bdgenomics.adam.algorithms.consensus.ConsensusGenerator.fromKnowns(knownIndels._jvmRdd)
+        consensusModel = self.sc._jvm.org.bdgenomics.adam.algorithms.consensus.ConsensusGenerator.fromKnownIndels(knownIndels._jvmRdd, 0)
         return AlignmentRecordDataset(self._jvmRdd.realignIndels(consensusModel,
                                                                  isSorted,
                                                                  maxIndelSize,

--- a/adam-python/bdgenomics/adam/test/alignmentRecordDataset_test.py
+++ b/adam-python/bdgenomics/adam/test/alignmentRecordDataset_test.py
@@ -359,3 +359,29 @@ class AlignmentRecordDatasetTest(SparkTestCase):
         jRdd = reads.rightOuterShuffleRegionJoinAndGroupByLeft(targets)
 
         self.assertEqual(jRdd.toDF().count(), 21)
+
+
+    def test_realignIndels_reads(self):
+
+        readsPath = self.resourceFile("small.1.sam")
+
+        ac = ADAMContext(self.ss)
+
+        reads = ac.loadAlignments(readsPath)
+        realigned = reads.realignIndels()
+
+        self.assertEqual(realigned.toDF().count(), 20)
+
+    def test_realignIndels_known_indels(self):
+
+        readsPath = self.resourceFile("small.1.sam")
+        variantsPath = self.resourceFile("small.vcf")
+
+        ac = ADAMContext(self.ss)
+
+        reads = ac.loadAlignments(readsPath)
+        knownIndels = ac.loadVariants(variantsPath)
+
+        realigned = reads.realignIndelsFromKnownIndels(knownIndels)
+
+        self.assertEqual(realigned.toDF().count(), 20)

